src/T2DMap.cpp

```diff
@@ -4114,15 +4114,35 @@ void T2DMap::slot_setRoomWeight()
                     itWeightsUsed.next();
                     if (itWeightsUsed.value() == weightCountsList.at(i)) {
                         if (itWeightsUsed.key() == 1) { // Indicate the "default" value which is unity weight
-                            displayStrings.append(tr("%1 {count:%2, default}").arg(QString::number(itWeightsUsed.key()), QString::number(itWeightsUsed.value())));
+                            displayStrings.append(tr("%1 {count:%2, default}",
+                                                     // Intentional comment to separate arguments
+                                                     "Everything after the first parameter (the '%1') will be removed by processing "
+                                                     "it as a QRegularExpression programmatically, ensure the translated text has "
+                                                     "` {` immediately after the '%1', and '}' as the very last character, so that the "
+                                                     "right portion can be extracted if the user clicks on this item when it is shown "
+                                                     "in the QComboBox it is put in.")
```

Comment:
These comments "for translators" use too many internal details, and too little actual explanation of words to translate.
```suggestion
                            displayStrings.append(tr("%1 {count:%2, default}",
                                                     // Intentional comment to separate arguments
                                                     "This text will be part of a list of room weights shown, where %1 will be "
                                                     "the number for the weight, and %2 counts the amount of times "
                                                     "this weight is used. Also this special weight is the default value. "
                                                     "When translating, ensure the curly braces start immediately after "
                                                     "the weight number, and the text must end with closing curly braces.")
```

Reply:
I'm not sure I follow you - surely it behoves the developers to explain what the application will be providing to go in as each of the %1, %2 ... %N parameter so the translators have the context, given that each %? thing could be literally *anything* (including BTW nothing at all!). In this particular case (and one or two others) the text that is provided in the localised case needs to be processed by the application itself ("is used programmatically") and thus the translation needs to be in a certain form in order to still work correctly in a language other than the "Engineering English".

However in this particular case I realise the code can be made more robust by merely ensuring that the first group of Arabic digits is considered as the weight to use and there is no need to specify anything more than that the translation does not begin with a number - rather than that it has the form (precisely as a *regexp*) of  **`" \{.*\}"`**.